tree/dataframe/inc/ROOT/RDF/RInterface.hxx

```diff
@@ -526,10 +525,9 @@ public:
                << RDFInternal::PrettyPrintAddr(&columnList) << "),"
                << "*reinterpret_cast<ROOT::RDF::RSnapshotOptions*>(" << RDFInternal::PrettyPrintAddr(&options) << "));";
       // jit snapCall, return result
-      TInterpreter::EErrorCode errorCode;
-      gInterpreter->Calc(snapCall.str().c_str(), &errorCode);
-      if (TInterpreter::EErrorCode::kNoError != errorCode) {
-         std::string msg = "Cannot jit Snapshot call. Interpreter error code is " + std::to_string(errorCode) + ".";
+      auto calcRes = RDFInternal::InterpreterCalc(snapCall.str());
+      if (0 != calcRes.second) {
+         std::string msg = "Cannot jit Snapshot call. Interpreter error code is " + std::to_string(calcRes.second) + ".";
```

Comment:
is this error message still accurate (here and below)? iiuc we return the exact interprete error code anymore

Reply:
Yes and maybe this is the best we can do now?

Reply:
(I know that this interpreter calls need to go, that is really the best we can do)

Reply:
we could remove "interpreter error code is ..." from the error message since the number reported is not what the interpreter reported

Reply:
now we do.